frontend/app/src/components/helper/CollectionHandler.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <div>
+    <slot
+      :data="data"
+      :limit="limit"
+      :found="found"
+      :total="total"
+      :total-usd-value="totalUsdValue"
+      :item-length="itemLength"
+      :show-upgrade-row="showUpgradeRow"
+    />
+  </div>
+</template>
+<script setup lang="ts">
+import { PropType } from 'vue';
+import { Collection } from '@/types/collection';
+import { getCollectionData, setupEntryLimit } from '@/utils/collection';
+
+const props = defineProps({
```

Comment:
This is not what I had in mind but we can keep it for now.

I was thinking more of creating a generic composable to handle the common logic with he options/pagination filter updating. But we can do it in the future no worries.